sygnal/helper/proxy/connectproxyclient_twisted.py

```diff
@@ -187,7 +194,7 @@ def connectionLost(self, reason=connectionDone):
         if not self.connected_deferred.called:
             self.connected_deferred.errback(reason)
 
-    def proxyConnected(self, _):
+    def proxyConnected(self, _: Union[None, "defer.Deferred[None]"]) -> None:
```

Comment:
```suggestion
    def proxyConnected(self, _: Optional["defer.Deferred[None]"]) -> None:
```

Might need to import Optional if it's not already imported in this file.

Reply:
Done. I thought it had a deeper purpose.
https://github.com/matrix-org/synapse/blob/aec69d2481e9ea1d8ea1c0ffce1706a65a7896a8/synapse/http/connectproxyclient.py#L213

Reply:
No deeper purpose here: `Optional[T]` is literally a alias of `Union[T, None]`. (In more modern versions of Python, it's spelt `T | None`, but the `|` operator for types wasn't introduced until Python 3.9, I think.)